### PR TITLE
added 23503 pgerrcode support and updated the error conversion approach

### PR DESCRIPTION
**Pull request by codingamir**
- [Y ] Do only one thing
- [ Y] Non breaking API changes
- [ Y] Tested

### What did this pull request do?

Added 23503 pgerrcode support and updated the error conversion from pg error to gorm error approach

### CODE REVIEW

File: error_translator.go [hunk 1]
```diff
@@ -5,14 +5,16 @@ import (
...
+var pgerrodesToGormErrors = map[string]error{
+	"23505": gorm.ErrDuplicatedKey,
+	"23503": gorm.ErrForeignKeyViolated,
 }
 
 func (dialector Dialector) Translate(err error) error {
 	if pgErr, ok := err.(*pgconn.PgError); ok {
-		if pgErr.Code == errCodes["uniqueConstraint"] {
-			return gorm.ErrDuplicatedKey
+		val, ok := pgerrodesToGormErrors[pgErr.Code]
```
Comment (a631807682) [inline error_translator.go:15]: rename `val`, possibly `gormErr`

**Comment by saeidee** [comment]
You may have conflict after merging this https://github.com/go-gorm/postgres/pull/179

File: error_translator.go [hunk 2]
```diff
@@ -5,14 +5,16 @@ import (
 	"gorm.io/gorm"
 )
 
-var errCodes = map[string]string{
-	"uniqueConstraint": "23505",
+var pgerrodesToGormErrors = map[string]error{
```
Comment (saeidee) [inline error_translator.go:8]: If you see here https://github.com/go-gorm/postgres/pull/179, these errors are not just for pg anymore, maybe we can rename this to `errDesToGormErrs`?

**Comment by saeidee** [comment]
@jinzhu could you please revert this, as it has not properly resolved the conflict with this https://github.com/go-gorm/postgres/pull/179
image

**Comment by watson-sam** [comment]
@saeidee just opened an issue (https://github.com/go-gorm/postgres/issues/187) as hadnt seen yout above comment but this is indeed causing an error

**Comment by jinzhu** [comment]
> @jinzhu could you please revert this, as it has not properly resolved the conflict with this #179 as well as needs to update gorm version to have the new error type. image

Can you submit the revert commit and release a new version?

(I haven't something wrong with my development machine  ;(

**Comment by saeidee** [comment]
@codingamir feel free to create a new PR & thanks for contributing.

**Pull request merged**